chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/ApiKeyActivityDAO.java

```diff
@@ -1,33 +1,113 @@
 package gov.healthit.chpl.dao;
 
+import java.util.Date;
 import java.util.List;
 
+import javax.persistence.Query;
+
+import org.springframework.stereotype.Repository;
+
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
 import gov.healthit.chpl.dto.ApiKeyActivityDTO;
 import gov.healthit.chpl.entity.ApiKeyActivityEntity;
 import gov.healthit.chpl.exception.EntityCreationException;
 import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.util.AuthUtil;
+
+@Repository("apiKeyActivityDAO")
+public class ApiKeyActivityDAO extends BaseDAOImpl {
+
+    public ApiKeyActivityDTO create(ApiKeyActivityDTO dto) throws EntityCreationException {
+        ApiKeyActivityEntity entity = null;
+        try {
+            if (dto.getId() != null) {
+                entity = this.getEntityById(dto.getId());
+            }
+        } catch (final EntityRetrievalException e) {
+            throw new EntityCreationException(e);
+        }
+
+        if (entity != null) {
+            throw new EntityCreationException("An entity with this ID already exists.");
+        } else {
+            entity = new ApiKeyActivityEntity();
+            entity.setApiKeyId(dto.getApiKeyId());
+            entity.setApiCallPath(dto.getApiCallPath());
+            entity.setApiCallMethod(dto.getApiCallMethod());
+
+            if (dto.getLastModifiedDate() != null) {
+                entity.setLastModifiedDate(dto.getLastModifiedDate());
+            } else {
+                entity.setLastModifiedDate(new Date());
+            }
+            if (dto.getCreationDate() != null) {
+                entity.setCreationDate(dto.getCreationDate());
+            } else {
+                entity.setCreationDate(new Date());
+            }
+            entity.setDeleted(dto.getDeleted());
+
+            if (AuthUtil.getCurrentUser() == null) {
+                entity.setLastModifiedUser(-3L);
```

Comment:
I think we can User.SYSTEM_USER_ID.

Reply:
Good catch. If you're okay with it, I'll make that change before this goes to STG, but not hold up the progress across the board based on that change. Suggest leaving this conversation unresolved until that change is made though